tests/queries/0_stateless/03004_force_null_for_omitted.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+# Tags: no-parallel
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT -q "insert into function file(03004_data.bsonEachRow, auto, 'null Nullable(UInt32)') select number % 2 ? NULL : number from numbers(5) settings engine_file_truncate_on_insert=1"
+$CLICKHOUSE_CLIENT -q "select * from file(03004_data.bsonEachRow, auto, 'null UInt32, foo UInt32')"
+$CLICKHOUSE_CLIENT -q "select * from file(03004_data.bsonEachRow, auto, 'null UInt32, foo UInt32') settings input_format_force_null_for_omitted_fields = 1" 2>&1 | grep -q -F "TYPE_MISMATCH" && echo "OK" || echo "FAIL"
+
+$CLICKHOUSE_CLIENT -q "select * from format(Values, 'foo UInt32', '()')"
+$CLICKHOUSE_CLIENT -q "select * from format(Values, 'foo UInt32', '()') settings input_format_force_null_for_omitted_fields = 1" 2>&1 | grep -q -F "TYPE_MISMATCH" && echo "OK" || echo "FAIL"
+
+$CLICKHOUSE_CLIENT -q "select * from format(JSONEachRow, 'foo UInt32', '{}')"
+$CLICKHOUSE_CLIENT -q "select * from format(JSONEachRow, 'foo UInt32', '{}') settings input_format_force_null_for_omitted_fields = 1" 2>&1 | grep -q -F "TYPE_MISMATCH" && echo "OK" || echo "FAIL"
+$CLICKHOUSE_CLIENT -q "select * from format(JSONEachRow, 'foo UInt32, bar Nullable(UInt32)', '{}')"
+$CLICKHOUSE_CLIENT -q "select * from format(JSONEachRow, 'foo UInt32, bar Nullable(UInt32)', '{\"foo\":1}')"
+$CLICKHOUSE_CLIENT -q "select * from format(JSONEachRow, 'foo UInt32, bar Nullable(UInt32)', '{}') settings input_format_force_null_for_omitted_fields = 1" 2>&1 | grep -q -F "TYPE_MISMATCH" && echo "OK" || echo "FAIL"
+$CLICKHOUSE_CLIENT -q "select * from format(JSONEachRow, 'foo UInt32, bar Nullable(UInt32)', '{\"foo\":1}') settings input_format_force_null_for_omitted_fields = 1"
```

Comment:
I would add tests like this to both `BSONEachRow` and `Values` formats to make sure reading is okay with the new setting.

I would also consider adding some tests with at least one type that `isLowCardinalityNullable`, both positive and negative cases.

Reply:
A lot of these tests do not make sense for `Values` format, because it's position-based.
I will add low cardinality ones though

Reply:
Added LowCardinality